components/CommentAdder.jsx

```diff
@@ -0,0 +1,52 @@
+import { UserContext } from "../context/User";
+import { useContext, useState } from "react";
+import { postComment } from "../Api";
+function CommentAdder({ setCurrentComments, article_id }) {
+  const { loggedInUser, setLoggedInUser } = useContext(UserContext);
+  const [commentInput, setCommentInput] = useState("");
+  function handleSubmit(event) {
+    event.preventDefault();
+    const body = {
+      username: loggedInUser.username,
+      body: commentInput,
+    };
+    setCurrentComments((currComments) => {
+      const newCommentBody = {
+        body: commentInput,
+        author: loggedInUser.username,
+        votes: 0,
+        comment_id: currComments.length + 1,
+      };
+      postComment(article_id, body).catch((err) => {
```

Comment:
Ideally here we want some logic to handle the scenario that the post fails, such as being given feedback on unsuccessful post